lib/format_routing_constraint.rb

```diff
@@ -0,0 +1,23 @@
+class FormatRoutingConstraint
+  def initialize(format, artefact_retriever = PreconfiguredArtefactRetriever)
```

Comment:
I don't know have loads of context so this is more of a 'how I read it' comment than a suggestion.

Passing in `artefact_retriever = PreconfiguredArtefactRetriever` confused me for a bit because it made me think `PreconfiguredArtefactRetriever` was a type of `ArtefactRetriever`, or at least had the same interface. The signatures of `#initialize` and `#fetch_artefact` are not quite compatible.

I dunno what might be better. Renaming? Making them a bit more polymorphic? Or maybe this is just a transitional thing?